# HG changeset patch
# User andrew
# Date 1405461315 -3600
#      Tue Jul 15 22:55:15 2014 +0100
# Node ID 967ae4d849eb20b4b63c399bc3f376dfc0f9d2cc
# Parent  84854c983fd81cf1a1598f8c069a2db010abc05c
PR1869: Avoid x86 workaround when running Zero rather than a JIT
Contributed-by: Matthias Klose <doko@ubuntu.com>

diff -Nru openjdk.orig/hotspot/src/os/linux/vm/os_linux.cpp openjdk/hotspot/src/os/linux/vm/os_linux.cpp
--- openjdk.orig/hotspot/src/os/linux/vm/os_linux.cpp	2019-04-11 03:36:49.000000000 +0100
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2019-04-19 17:15:35.194017652 +0100
@@ -5139,7 +5139,7 @@
 
   Linux::capture_initial_stack(JavaThread::stack_size_at_create());
 
-#if defined(IA32)
+#if defined(IA32) && !defined(ZERO)
   workaround_expand_exec_shield_cs_limit();
 #endif
 
diff -Nru openjdk.orig/hotspot/src/os_cpu/linux_x86/vm/os_linux_x86.cpp openjdk/hotspot/src/os_cpu/linux_x86/vm/os_linux_x86.cpp
--- openjdk.orig/hotspot/src/os_cpu/linux_x86/vm/os_linux_x86.cpp	2019-04-11 03:36:49.000000000 +0100
+++ openjdk/hotspot/src/os_cpu/linux_x86/vm/os_linux_x86.cpp	2019-04-19 17:15:58.329656663 +0100
@@ -890,7 +890,7 @@
  * updates (JDK-8023956).
  */
 void os::workaround_expand_exec_shield_cs_limit() {
-#if defined(IA32)
+#if defined(IA32) && !defined(ZERO)
   size_t page_size = os::vm_page_size();
 
   /*
